HeatApp/components/screens/UserSettingsScreen.js

```diff
@@ -14,27 +16,202 @@ export default class UserSettingsScreen extends React.Component {
     )
   };
 
+  genderMatch = {
+    1: 'Male',
+    2: 'Female',
+    3: 'Other'
+  };
+
   constructor(props) {
     super(props);
+    this.state = {
+      gender: 3,
+    };
+  }
+
+  getStateClone() {
+    var stateClone = {};
+    Object.keys(this.state).forEach(key => {
+      stateClone[key] = this.state[key];
+    });
+    return stateClone;
+  }
+
+  updateState(field, value) {
+    var newState = this.getStateClone();
+    newState[field] = value;
+    this.setState(newState);
+  }
+
+  updateUsername(text) {
+    this.updateState('username', text);
+  }
+
+  updatePassword(text) {
```

Comment:
What about arrow-expressions like `updatePassword = (text) => {...}`? You can call it directly from elements without binding it.

Reply:
Done.